biom/commands/table_validator.py

```diff
@@ -72,7 +72,11 @@ class TableValidator(Command):
     HDF5FormatVersions = set([(2, 0), (2, 0, 0), (2, 1), (2, 1, 0)])
 
     def run(self, **kwargs):
-        is_json = kwargs['is_json']
+        # We can't trust kwargs['is_json'] because that's determined
+        # before any parsing happens...
+        with open(kwargs['table']) as f:
```

Comment:
We are doing the same in `biom_open` so I'm inclined to say that we should move these two lines into a function living in `util.py`, maybe `is_hdf5_file`? Depending on what you think, a test should also be added.